crates/new_task/src/dependency_task.rs

```diff
@@ -0,0 +1,129 @@
+#[cfg(async_executor_impl = "async-std")]
+use async_std::task::{spawn, JoinHandle};
+#[cfg(async_executor_impl = "tokio")]
+use tokio::task::{spawn, JoinHandle};
+
+use futures::Future;
+
+use crate::dependency::Dependency;
+
+pub trait HandleDepResult: Send + Sized + Sync + 'static {
+    type Result: Send + Sync + 'static;
+
+    /// Called once when the Dependency completes handles the results
+    fn handle_dep_result(self, res: Self::Result) -> impl Future<Output = ()> + Send;
+}
+
+pub struct DependencyTask<D: Dependency<H::Result> + Send, H: HandleDepResult + Send> {
```

Comment:
Nit: did we include this crate in the lints?  I feel like clippy should be warning for no documentation.

Reply:
Hmm it must not be enabled, I'll clean up